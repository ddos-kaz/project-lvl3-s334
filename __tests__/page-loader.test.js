import axios from 'axios';
import httpAdapter from 'axios/lib/adapters/http';
import nock from 'nock';
import os from 'os';
import fs from 'mz/fs';
import path from 'path';
import loader from '../src';

describe('Page Loader', () => {
  const host = 'http://localhost';
  const pathname = '/test/second';
  axios.defaults.host = host;
  axios.defaults.adapter = httpAdapter;
  beforeAll(async () => {
    const testData = await fs.readFile(path.resolve(__dirname, '__fixtures__', 'simple_page.html'), 'utf8');
    nock(host).get(pathname).reply(200, testData);
  });
  it('Simple loading page content', async () => {
    const fileName = 'localhost-test-second.html';
    const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), 'foo-'));
    await loader(`${host}${pathname}`, tempDir);
    const responseData = await fs.readFile(path.resolve(tempDir, fileName), 'utf8');
    axios.get(`${host}${pathname}`).then((res) => {
      expect(responseData).toEqual(res.data);
    }).catch(err => console.log(err));
  });
});
