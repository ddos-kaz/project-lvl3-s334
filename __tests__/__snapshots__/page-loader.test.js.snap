// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Loader Status code is not 201 1`] = `"Expected response code '200', but current code is '201' for 'http://localhost/second'"`;

exports[`Page Loader Transforming html file and downloading resources 1`] = `
"<html><head>
     <script src=\\"localhost-test_files/src-script.js\\"></script>
     <link rel=\\"stylesheet\\" href=\\"localhost-test_files/src-style.css\\">
  </head>
  <body>
    <h1>My First Heading</h1>
    <p>My first paragraph.</p>
    <img src=\\"localhost-test_files/src-hexlet_logo.png\\" alt=\\"hexlet-logo\\" height=\\"200\\" width=\\"200\\">
    <img src=\\"localhost-test_files/developer-cdn-mozilla-net-media-mdn-logo-sm.png\\" alt=\\"MDN logo &#x2014; a dinosaur logo with the text MDN\\">
    <img>
  

</body></html>"
`;

exports[`Page Loader Wrong output directory 1`] = `"Passed '/var/tmp/test' does not exist!"`;
